src/web/controllers/admin/player_admin_controller.py

```diff
@@ -1226,10 +1226,12 @@ async def htmx_admin_player_move(
         admin_player = web_context.get_admin_player()
         dst_tournament = web_context.get_admin_tournament()
         src_tournament = admin_player.tournament
+        admin_event = web_context.get_admin_event()
         try:
             self._validate_player_tournament_move(admin_player, dst_tournament)
-            src_tournament.delete_player_from_tournament(admin_player.id)
-            dst_tournament.add_player_to_tournament(admin_player.stored_player)
+            admin_event.move_player_to_tournament(
+                admin_player, src_tournament, dst_tournament
+            )
```

Comment:
```suggestion
            dst_tournament.add_player_to_tournament(admin_player.stored_player)
            src_tournament.delete_player_from_tournament(admin_player.id)
```
This is all that is needed, we need a fresh tournament_player record when changing a player's tournament, because the values other than `tournament_id` will differ. For now, it means the player keeps its pairing number, but from what is planned tournament_player will contain all the player values specific to the tournament.

Reply:
There is a subtle problem with your approach, in that it can duplicate players if something goes wrong, while it doesn't in mine.
I don't see why moving players between tournaments needs to keep being a tournament-level operation instead of an event-level operation, since all the values are inside the event database

Reply:
how could the player be duplicated ? all it does is delete the tournament_player that exists and create a new one, I don't how it would be more dangerous than yours. Anything that handles `player` is at event level, but this handles `tournament_player`, which is something at tournament level

Reply:
you could argue that the controller shouldn't be responsible for knowing _how_ to move players, that should be an atomic operation in EventDatabase.

Reply:
```py
            self._validate_player_tournament_move(admin_player, dst_tournament)
            dst_tournament.add_player_to_tournament(admin_player.stored_player)
            src_tournament.delete_player_from_tournament(admin_player.id)
```

If you wanna put these lines into an `Event.move_player` be my guest. not sure it's worth arguing about as moving player will most likely disappear with multi-tournament, the important part was to fix the bug which is done

Reply:
> how could the player be duplicated ? all it does is delete the tournament_player that exists and create a new one, I don't how it would be more dangerous than yours. Anything that handles `player` is at event level, but this handles `tournament_player`, which is something at tournament level

Add a player to the destination tournament, crash before deleting from the source tournament, now the player is assigned to two tournaments